### PR TITLE
14.0 mig website portal contact

### PR DESCRIPTION
**Pull request by chandni299**
Migration of module "website_portal_contact" to v14

### CODE REVIEW

**Comment by pedrobaeza** [comment]
I think this module needs at least a renaming, as no more dependency on website is needed for handling the portal. Thus, it can be moved to OCA/partner-contact.

**Comment by chandni299** [comment]
> I think this module needs at least a renaming, as no more dependency on website is needed for handling the portal. Thus, it can be moved to OCA/partner-con

Thanks ! Then what should be the new name or should I wait for a reply from Tecnativa's suggestion?

**Comment by pedrobaeza** [comment]
Well, I'm from Tecnativa :wink: 

I good new name can be `portal_contact_mgmt`.

**Review by claudiagn — COMMENTED** [review]
I also think this should be moved to contract! 👍

**Review by KilianNV — APPROVED** [review]
LGTM

File: website_portal_contact/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+{
+    "name": "Contact Manager In Website Portal",
+    "summary": "Allows logged in portal users to manage their contacts",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "category": "Portal",
+    "website": "https://tecnativa.com/",
```
Comment (MeetKD) [inline website_portal_contact/__manifest__.py:9]: Website should be as per oca guideline.

File: website_portal_contact/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        """Get user's contacts domain."""
+        partner = request.env.user.partner_id
+        Partner = request.env["res.partner"]
+        domain = [("type", "=", "contact"), ("active", "=", True)]
+        domain += request.env.ref(
+            "website_portal_contact.rule_edit_own_contacts"
+        )._compute_domain("res.partner", "read")
+
+        # To edit yourself you have /my/account
+        domain += [("id", "!=", request.env.user.partner_id.id)]
```
Comment (MeetKD) [inline website_portal_contact/controllers/main.py:37]:
```suggestion
        domain += [("id", "!=", partner.id)]
```

File: website_portal_contact/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+        values = super()._prepare_home_portal_values(counters)
+        Partner = request.env["res.partner"]
+        if "contact_count" in counters:
+            values["contact_count"] = Partner.search_count(self._contacts_domain())
+        return values
+
+    def _contacts_domain(self):
+        """Get user's contacts domain."""
+        partner = request.env.user.partner_id
+        Partner = request.env["res.partner"]
```
Comment (MeetKD) [inline website_portal_contact/controllers/main.py:30]: Do we need this variable?

File: website_portal_contact/views/assests.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+     License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). -->
+<odoo>
+
+    <template id="assets_frontend" inherit_id="website.assets_frontend" name="Wishlist assets frontend">
```
Comment (MeetKD) [inline website_portal_contact/views/assests.xml:6]:
```suggestion
    <template id="assets_frontend" inherit_id="website.assets_frontend" name="contact assets frontend">
```

File: website_portal_contact/views/contact_portal_template.xml [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                <t t-esc="contact.name"/>
+            </li>
+            <li t-if="page_name == 'add_contact' or add_contact" t-attf-class="breadcrumb-item active">
+                <t>New Contact</t>
+            </li>
+        </xpath>
+    </template>
+
+
+
```
Comment (MeetKD) [inline website_portal_contact/views/contact_portal_template.xml:33]: Can you please remove extra spaces?

File: website_portal_contact/views/contact_portal_template.xml [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+                </t>
+                <t t-set="card_body">
+                    <form id="portal_contact"
+                        t-attf-action="/my/contacts/{{'{}/update'.format(contact.id) if contact.id else 'create'}}"
+                        method="post"
+                        enctype="multipart/form-data"
+                        class="container form-horizontal">
+                        <input type="hidden" name="csrf_token" t-att-value="request.csrf_token()"/>
+                        <t t-foreach="fields" t-as="field_name">
+                            <!-- <t t-call="website_portal_contact.form_field"/> -->
```
Comment (MeetKD) [inline website_portal_contact/views/contact_portal_template.xml:126]: Please remove commented code.

**Review by MeetKD — CHANGES_REQUESTED** [review]

**Comment by chandni299** [comment]
Hello @MeetKD , You will find the correct PR to review the code here for v14.
where now Migration of module "website_portal_contact" to v14 is moved to OCA/partner-contact with new module name "portal_contact_mgmt"
 [OCA/partner-contact/PR-1057](https://github.com/OCA/partner-contact/pull/1057)

**Pull request closed without merging**